docs/doxygen/generate_docs.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
```

Comment:
Should this file be in `tools/` instead of `docs/`?

Reply:
Was pondering this last night.